src/marks/auto.d.ts

```diff
@@ -3,26 +3,208 @@ import type {CompoundMark, Data} from "../mark.js";
 import type {Reducer} from "../reducer.js";
 import type {BinOptions} from "../transforms/bin.js";
 
+/**
+ * One of the categories of mark; *line* includes lineX and lineY; *area*
+ * includes areaX and areaY; *rule* includes ruleX and ruleY; *bar* includes
+ * barX, barY, rectX, and rectY; and *dot* is just dot.
+ */
+type MarkType = "dot" | "line" | "area" | "rule" | "bar";
```

Comment:
I’m not 100% sure when we should break enums out like this, versus leaving them in-place within the interface. But I think in this case it feels like it should be inlined in AutoOptions. Note that we can use AutoOptions["mark"] to still refer to this type in another place (which I did for AutoSpec).